api/recommend.js

```diff
@@ -1,12 +1,67 @@
 const express = require('express');
 const asyncHandler = require('express-async-handler');
-const recombee = require('recombee-api-client');
 const crypto = require('crypto');
+const recombee = require('recombee-api-client');
 
 const rqs = recombee.requests;
 const recombeeClient = new recombee.ApiClient(process.env.RECOMBEE_DATABASE_ID,
   process.env.RECOMBEE_PRIVATE_TOKEN);
 
+async function recPersonal(userHash, count) {
+  let result;
+  await recombeeClient.send(
+    new rqs.RecommendItemsToUser(userHash, count, { scenario: 'personal_view' }),
+  )
+    .then((recommendation) => {
+      result = { recipes: recommendation.recomms || [] };
+    })
+    .catch(() => {
+      recombeeClient.send(
+        new recombee.RecommendItemsToUser(userHash, count, { scenario: 'homepage_view' }),
+      );
+    })
+    .then((recommendation) => {
+      result = { recipes: recommendation.recomms || [] };
+    })
+    .catch((err) => {
+      throw (err);
+    });
+  return result;
+}
+
+async function recPopular(userHash, count) {
+  let result;
+  await recombeeClient.send(
+    new rqs.RecommendItemsToUser(userHash, count, { scenario: 'popular_view' }),
+  )
+    .then((recommendation) => {
+      result = { recipes: recommendation.recomms || [] };
+    })
+    .catch(() => { result = { recipes: [] }; })
+    .catch((err) => {
+      throw (err);
+    });
+  return result;
+}
+
+
+async function recExplore(userHash, count) {
+  const result = { personal: [], popular: [] };
+
+  await recPersonal(userHash, count)
+    .then((res) => { result.personal = res; })
+    .catch((err) => {
+      throw (err);
```

Comment:
this is not the behaviour we want. if you can't get personal recommendation then don't set result.perosnal but continue to server the request.